minio/policy.py

```diff
@@ -240,8 +240,8 @@ def _remove_write_only(actions, cond_value):
 
     actions = _get_action(statement)
     cond_value = statement.get('Condition', {})
-    if (statement.get('Effect') == 'Allow' and
-            statement.get('Principal') == {'AWS': '*'}):
+    aws = statement.get('Principal', {}).get('AWS', [])
+    if (statement.get('Effect') == 'Allow' and (aws == '*' or '*' in aws) ):
```

Comment:
```(aws == '*' or '*' in aws) ```can be replaced by ```('*' in aws)``` everywhere this comparison is done.

Reply:
Our requirement is to get `True` for cases like `aws` is a string and strictly set to `'*'`, or `aws` is a list and at least one of its element is exactly`'*'`, like`['abc, ..., '*', ..., 'def']`. So, we need to reject or return `False` for string values like `'abc*'`.
If we drop `aws == '*'` part of the condition, then `('*' in aws)` will return `True` for cases like `'abc*'`, and we don't want that. That's why we have to have both of the conditions, `(aws == '*' or '*' in aws)`, in place.

Reply:
ok, makes sense.